File: src/main/java/com/javawarriors/buyerside/services/BuyerQnAService.java
package com.javawarriors.buyerside.services;

import com.javawarriors.buyerside.entities.*;
import com.javawarriors.buyerside.repositories.*;
import com.javawarriors.buyerside.entities.compositeKeys.DealPK;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.*;

/**
 * Service layer for handling logic related to buyer QnA
 */
@Service
public class BuyerQnAService {
    @Autowired
    private BuyerQnARepo buyerQnARepo;
    @Autowired
    private AnswerQnARepo answerQnARepo;

    @Autowired
    private WTBService wtbService;
    @Autowired
    private DealService dealService;

    public List<BuyerQnA> saveManyBuyerQnAs(Iterable<BuyerQnA> buyerQnAs) {
        return buyerQnARepo.saveAll(buyerQnAs);
    }

    public List<BuyerQnA> findByWtbId(Long wtbId) {
        WantToBuyListing wtbListing = wtbService.findById(wtbId);
        return buyerQnARepo.findByWtbListing(wtbListing);
    }

    public void deleteByWtbId(Long wtbId) {
        List<BuyerQnA> toDeleteQnA = findByWtbId(wtbId);
        for (BuyerQnA buyerQnA : toDeleteQnA) {
            buyerQnARepo.delete(buyerQnA);
        }
    }

    public List<AnswerQnA> saveManyAnswerQnAs(Iterable<AnswerQnA> answerQnAs) {
        return answerQnARepo.saveAll(answerQnAs);
    }

    public List<AnswerQnA> findAnswerQnAByDeal(Long sellerId, Long wtbId, Long ifsId) {
        DealPK dealPK = new DealPK(sellerId, wtbId, ifsId);
        Deal deal = dealService.findById(dealPK);
        return answerQnARepo.findByDeal(deal);

    }
}

File: src/main/frontend/src/services/BuyerQnAService.js
import axios from "axios";
import authHeader from "../util/authHeader";
import { API_BASE_URL } from "../util/apiBaseUrl";

const BUYER_QNA_API_BASE_URL = `${API_BASE_URL}/buyer-qna`;

class BuyerQnAService {
  postManyBuyerQnAs(buyerQnAs) {
    return axios.post(BUYER_QNA_API_BASE_URL + "/post/many", buyerQnAs, {
      headers: authHeader(),
    });
  }

  getQnAsForWtbListing(wtbId) {
    return axios.get(BUYER_QNA_API_BASE_URL + "/get/wtblisting=" + wtbId, {
      headers: authHeader(),
    });
  }

  postManyAnswerQnAs(answerQnAs) {
    return axios.post(
      BUYER_QNA_API_BASE_URL + "/post/many-answers",
      answerQnAs,
      {
        headers: authHeader(),
      }
    );
  }

  getAnswersByDeal(sellerId, wtbId, ifsId) {
    return axios.get(
      BUYER_QNA_API_BASE_URL +
        "/get/answer/seller=" +
        sellerId +
        "&wtbid=" +
        wtbId +
        "&ifsid=" +
        ifsId,
      {
        headers: authHeader(),
      }
    );
  }
}

export default new BuyerQnAService();
